messaging/src/main/java/org/axonframework/eventhandling/EventHandlerInvoker.java

```diff
@@ -92,4 +92,13 @@ default <R> void performReset(@Nullable R resetContext) {
             );
         }
     }
+
+    /**
+     * Should be implemented to clear a cache when processing of a segment is stopped. As another instance might start
+     * processing the segment, we can't trust the local cache state anymore for that segment.
+     *
+     * @param segmentId the id of the segment for which the cache needs to be cleared.
+     */
+    default void clearCache(int segmentId) {
```

Comment:
I fully comprehend the necessity of this operation, but I am uncertain whether I fancy this method's name.
Differently put, I am not convinced that the `EventHandlerInvoker` should expose cache-specific tasks.

Perhaps something more generic like `clear` or `reset` (not an option due to the other reset methods) to support more than just cache clearing feels for.
Or something aligning with what we're informing the invoker about, like `segmentTransfer` or `segmentReassigned`, might work better.

What do you think, @gklijs?

Reply:
Renamed to `segementReleased` and it's now called in the same transaction as the one releasing the claim on the token.